src/chariot/core/database/SqliteDatabase.py

```diff
@@ -0,0 +1,62 @@
+import sqlite3
+import os
+import csv
+
+from chariot.core.database.Database import Database
+
+
+class SqliteDatabase(Database):
+    """
+    Implements the Database parent class.
+    Creates a Sqlite database and writes to it.
+
+    Usage:
+    sqliteDB = SqliteDatabase()
+    sqliteDB.insertRow(1000, "json")
+    """
+
+    def __init__(self, db_path: str = 'database.db', flush: bool = False):
+        Database.__init__(self, db_path=db_path, flush=flush)
+        self.connectToDB()
+        if self.flush:
+            self.flushDB()
+
+    def initializeDB(self):
+        cursor = self.conn.cursor()
+        cursor.execute(
+            'CREATE TABLE IF NOT EXISTS IOTD('
+            'id INTEGER PRIMARY KEY, ' +
+            'abs_time DATETIME DEFAULT CURRENT_TIMESTAMP, ' +
+            'relative_time BIGINT, ' +
+            'freeform TEXT)')
+        self.conn.commit()
+
+    def connectToDB(self):
```

Comment:
there should be some return value to indicate that connection was successful. This will also impact the interface, so apply the same return value on that. I suggest a simple boolean and use the python annotation:
def connectToDB(self) -> bool :

Reply:
Might be more helpful to return the error message instead, or just try/catch and propagate the error.

Reply:
Eni, I think that youre right the better approach is to return an error and let it propogate

Reply:
Database code is no longer in the request, resolving this conversation.